source/ossec_elk_upgrade.rst

```diff
@@ -0,0 +1,96 @@
+
+Backing up
+----------
+
+Make sure you back up the following files, so that you won't lose your valuable config files. Make backups of the current Logstash, Elasticsearch and Kibana configuration files.
+Here's a list of most important files:
+
+1. the Logstash-Server config is located in /etc/logstash/conf.d/01-ossec.conf
+2. *most importantly*: backup the certificate and the key!
+3. the certificate is located in /etc/logstash/logstash-forwarder.crt (if you're running an older version the location could be /etc/pki/tls/certs/logstash-forwarder.crt)
+4. the key is located in /etc/logstash/logstash-forwarder.key (if you're running an older version the location could be /etc/pki/tls/private/logstash-forwarder.key)
+5. backup the elasticsearch template for logstash. /etc/logstash/elastic-ossec-template.json
+
+Updating Logstash
+^^^^^^^^^^^^^^^^^
+
+The following steps describe how to update the Logstash server:
+
+1. Update the Yum Repo with the latest logstash version. Current version recommended by WAZUH is Logstash 2.1
+2. run “yum update logstash”
+3. Clone the latest stable version of the WAZUH ossec repository from https://github.com/wazuh/ossec-wazuh into the home directory
+4. Copy the new 01-ossec.conf (Logstash Config File) into /etc/logstash/conf.d/ from the cloned git repository.
+5. Copy the backed up Key & Certificate into /etc/logstash
+6. Copy the new Elasticsearch template for Logstash into /etc/logstash/ from the cloned git repository.
+7. Start the Logstash service & verify that the logstash process is running and check the logstash log /var/log/logstash/logstash.log
+
+Updating Elastisearch
+^^^^^^^^^^^^^^^^^^^^^
+
+The following steps seem a bit tricky at first, please pay attention to the backup steps, especially when backing up the Elasticsearch Index.
+
+1. Backup the elasticsearch config yaml files: elastisearch.yml and logging.yml
+2. relevant lines in the elasticsearch.yml are:
+   cluster.name: ossec
+   node.name: ossec_node1
+3. **Important: Backup the Elasticsearch Index** everything in /var/lib/elasticsearch/ossec/*
+4. stop the Elasticsearch service
+5. Update the Yum Repo with the latest Elasticsearch version. Current version recommended by WAZUH is Elasticsearch
+6. run “yum update elasticsearch”
+7. Replace the elasticsearch.yml file with the new. Check if the relevant lines from above are still there (cluster.name and node.name)
+8. Verify that Elasticsearch is running by issuing in your terminal:
+   curl -XGET localhost:9200
+   curl -XGET 'http://localhost:9200/_cluster/health?pretty=true'
+
+finally run: ~/ossec-wazuh/extensions/elasticsearch/ && curl -XPUT “http://localhost:9200/_template/ossec/” -d “@elastic-ossec-template.json”
```

Comment:
cd ~/ossec ?
please use \` sign to escape the code
and this doublequotes won't work in console